### PR TITLE
feat(utxo-lib): add UtxoPsbt w/BitGO P2TR signing

### PR DESCRIPTION
**Pull request by brandonblack**
* Extends upstream PSBT class
* Tests assert that resulting signed transactions are identical between PSBT and TransactionBuilder
* Only supports BitGo style 2-of-2 script path signing

Issue: BG-47820

Co-authored-by: davidkaplan@bitgo.com

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/utxo-lib/src/bitgo/Unspent.ts
Comment (sangaman) [inline modules/utxo-lib/src/bitgo/Unspent.ts:140]: Why is this `- 1` and not `+ 1`? Since we're adding an input to a tx, I'd expect that input would increase the input count and go at the end of the index.

File: modules/utxo-lib/src/bitgo/Unspent.ts
Comment (sangaman) [inline modules/utxo-lib/src/bitgo/Unspent.ts:160]: In the case of a segwit unspent, we're calling `updateInput` twice. Once on line 142 and then again either here or in the `else` block for non-p2tr. Is there a reason we have to call it twice? Or can it be in one call? Where we'd check for either non-segwit, segwit p2tr, or segwit non-p2tr and have one updateInput call for each.

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts
Comment (sangaman) [inline modules/utxo-lib/src/bitgo/UtxoPsbt.ts:66]: Why is the casting necessary here?

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts
Comment (sangaman) [inline modules/utxo-lib/src/bitgo/UtxoPsbt.ts:71]: Does this apply to both methods below? What code changed? Is it possible to just call these mostly copied methods directly since we're already doing `extends Psbt`. In the case of `validateSignaturesOfAllInputs` below for example I'm thinking we should be able to call `super()` and pass in the hardcoded validator we want.

File: modules/utxo-lib/src/bitgo/transaction.ts
Comment (sangaman) [inline modules/utxo-lib/src/bitgo/transaction.ts:108]: Is it safe to be setting version to `undefined` in the case that it's not specified in the params? Or should we default `version = 2` in case it's not provided?

File: modules/utxo-lib/src/bitgo/wallet/chains.ts
Comment (sangaman) [inline modules/utxo-lib/src/bitgo/wallet/chains.ts:110]: It might make sense to include the non-script `p2wpkh` type as well here just for completeness sake.

File: modules/utxo-lib/src/noble_ecc.ts
Comment (sangaman) [inline modules/utxo-lib/src/noble_ecc.ts:71]: What's the purpose of this change?

**Review by sangaman — COMMENTED** [review]
Looks good overall but I made a handful of comments and questions.

File: modules/utxo-lib/src/bitgo/wallet/chains.ts
Comment (sangaman) [reply modules/utxo-lib/src/bitgo/wallet/chains.ts:110]: Never mind I realized this is using our bitgo chaincodes and we don't have one for p2wpkh since we don't use it, so disregard.

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts
Comment (brandonblack) [reply modules/utxo-lib/src/bitgo/UtxoPsbt.ts:66]: Because the `unsignedTx` is an `bip174.Transaction` interface, but we need to tell it what concrete implementation `bitcoinjs-lib.PsbtTransaction` we were using to get the `tx` off of it.

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts
Comment (brandonblack) [reply modules/utxo-lib/src/bitgo/UtxoPsbt.ts:71]: No, just this method. There is no Taproot support upstream. We can't call super because the Taproot fields are totally disjoint from the pre-Taproot fields. The validator just checks the sigs, but the sigs have to be fetched from an entirely other place.

File: modules/utxo-lib/src/bitgo/transaction.ts
Comment (brandonblack) [reply modules/utxo-lib/src/bitgo/transaction.ts:108]: on line 89, `getDefaultTransactionVersion`, and we throw in the switch if it's not set to expected values.

I think there is a bug here that we say that we don't support zcash, but we do. Will have to look into that.

File: modules/utxo-lib/src/noble_ecc.ts
Comment (brandonblack) [reply modules/utxo-lib/src/noble_ecc.ts:71]: Fixes an error caused by `ECPair`'s (strongly against my preferences) `testecc` changes. Could use a comment.

File: modules/utxo-lib/src/bitgo/Unspent.ts
Comment (brandonblack) [reply modules/utxo-lib/src/bitgo/Unspent.ts:160]: Could do it in one call, but it's more appropriate to do it in separate calls, because of the logic flow.

"Any segwit gets a witnessUtxo, taproot gets taproot fields, segwitv0 gets witnessScript"

`updateInput` is intended to be called many times, often by separate updaters (eg. one might have access to a full node to fetch UTXOs, and another might have access to pubkeys to generate scripts).

File: modules/utxo-lib/src/bitgo/Unspent.ts
Comment (brandonblack) [reply modules/utxo-lib/src/bitgo/Unspent.ts:140]: We _just_ added it, so the index (0-base) of the one we just added is the count (1-base) minus one.

File: modules/utxo-lib/package.json
Comment (mmcshinsky-bitgo) [inline modules/utxo-lib/package.json:54]: Is it possible to get another publish for this?

File: modules/utxo-lib/package.json
Comment (mmcshinsky-bitgo) [inline modules/utxo-lib/package.json:60]: Is there a todo to revert when this PR is merged and published? https://github.com/bitcoinjs/ecpair/pull/11

File: modules/utxo-lib/package.json
Comment (brandonblack) [reply modules/utxo-lib/package.json:54]: Yes, pending a little PR merge there, we will publish.

File: modules/utxo-lib/package.json
Comment (brandonblack) [reply modules/utxo-lib/package.json:60]: Yep, we're discussing the options there, as the maintainer has been idle for some months. May end up having to fork off a BitGo fork.

**Review by sangaman — APPROVED** [review]
LGTM pending questions on dependencies/forks.

**Review by mmcshinsky-bitgo — APPROVED** [review]

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/UtxoPsbt.ts:269]: unused - what was the intended purpose?

Comment (brandonblack) [reply modules/utxo-lib/src/bitgo/UtxoPsbt.ts:269]: Exists because it exists on the upstream equivalent method. Would have to go to the upstream `getHashForSig` and read the code to remember why it exists there.

**Pull request merged**
